aptos-move/framework/aptos-framework/sources/nonce_validation.move

```diff
@@ -0,0 +1,256 @@
+module aptos_framework::nonce_validation {
+    use aptos_std::table::{Self, Table};
+    use aptos_std::timestamp;
+    use aptos_std::big_ordered_map::{Self, BigOrderedMap};
+    use aptos_std::aptos_hash::sip_hash_from_value;
+    use aptos_std::error;
+    friend aptos_framework::genesis;
+    friend aptos_framework::transaction_validation;
+
+    const NUM_BUCKETS: u64 = 50000;
+
+    // After a transaction expires, we wait for GARBAGE_COLLECTION_INTERVAL_SECS_AFTER_TXN_EXPIRATION seconds
+    // before garbage collecting the transaction from the nonce history.
+    // We maintain an invariant that two transactions with the same (address, nonce) pair cannot be stored
+    // in the nonce hsitory if their transanction expiration times are less than
+    // `GARBAGE_COLLECTION_INTERVAL_SECS_AFTER_TXN_EXPIRATION` seconds apart.
+    const GARBAGE_COLLECTION_INTERVAL_SECS_AFTER_TXN_EXPIRATION: u64 = 65;
```

Comment:
rename back or rename to NONCE_REPLAY_PROTECTION_OVERLAP_INTERVAL_SECS or similar, because we are using it not just for GC, but for distance between nonces. (even if it it wasn't GCed)

Reply:
Renamed to NONCE_REPLAY_PROTECTION_OVERLAP_INTERVAL_SECS